evm_loader/cli/src/account_storage.rs

```diff
@@ -462,44 +462,19 @@ impl<'a> AccountStorage for EmulatorAccountStorage<'a> {
         self.block_timestamp.try_into().unwrap()
     }
 
-    fn block_hash(&self, number: U256) -> [u8; 32] {
-        info!("block_hash {number}");
-
-        let number = number.as_u64();
+    fn block_hash(&self, slot: u64) -> [u8; 32] {
+        info!("block_hash {slot}");
 
         self.add_solana_account(slot_hashes::ID, false);
-        self.add_solana_account(recent_blockhashes::ID, false);
-
-        if self.block_number <= number {
-            return <[u8; 32]>::default();
-        }
 
         if let Ok(Some(slot_hashes_account)) = self.get_account(&slot_hashes::ID) {
-            if let Ok(Some(recent_blockhashes_account)) = self.get_account(&recent_blockhashes::ID)
-            {
-                let slot_hashes_data = slot_hashes_account.data;
-                let slot_hashes_len = u64::from_le_bytes(slot_hashes_data[..8].try_into().unwrap());
-                for i in 0..slot_hashes_len {
-                    let offset = usize::try_from((i * 40) + 8).unwrap();
-                    let slot =
-                        u64::from_le_bytes(slot_hashes_data[offset..][..8].try_into().unwrap());
-                    if number == slot {
-                        let recent_blockhashes_data = recent_blockhashes_account.data;
-                        if offset + 32 > recent_blockhashes_data.len() {
-                            break;
-                        }
-                        return recent_blockhashes_data[offset..][..32].try_into().unwrap();
-                    }
-                }
-            }
-        }
-
-        if let Ok(timestamp) = self.config.rpc_client.get_block(number) {
-            let hash = bs58::decode(timestamp.blockhash).into_vec().unwrap();
-            hash.try_into().unwrap()
+            let slot_hashes_data = slot_hashes_account.data.as_slice();
+            find_slot_hash(slot, slot_hashes_data)
         } else {
-            warn!("Got error trying to get block hash");
-            generate_fake_block_hash(number)
+            panic!(
+                "Trying to get slot hash info without providing sysvar account: {}",
+                slot_hashes::ID
```

Comment:
This error message will be propagated to end user.
It does not make sense in emulator context and does not reflect what really happens

Reply:
Fixed